library/src/main/java/io/ona/kujaku/views/KujakuMapView.java

```diff
@@ -1293,5 +1316,123 @@ private void resetRejectionDialogContent() {
     public MapboxLocationComponentWrapper getMapboxLocationComponentWrapper() {
         return mapboxLocationComponentWrapper;
     }
+
+
+    /************** Tracking Service ***************/
+
+    /**
+     * Rebind to a running TrackingService instance
+     *
+     * @param context
+     * @param listener
+     */
+    public void resumeTrackingService(Context context, TrackingServiceListener listener) {
+        // TrackingService reconnection if connection was lost
+        if (! trackingServiceBound && TrackingService.isRunning()) {
+            this.trackingServiceListener = listener;
+            TrackingService.bindService(context, TrackingService.getIntent(context, null,null), connection);
+        }
+    }
+
+    /**
+     * Start TrackingService
+     *
+     * @param context
+     * @param cls
+     * @param trackingServiceListener
+     * @param options
+     */
+    public void startTrackingService(@NonNull Context context, @NonNull Class<?> cls, @NonNull TrackingServiceListener trackingServiceListener, TrackingServiceOptions options) {
+        this.trackingServiceListener = trackingServiceListener;
```

Comment:
Can you request for Storage permissions at this point? Or whenever they are needed. They are not requested by default in this view? They are only requested in the sample application.

Reply:
I was thinking that the sample app is in charge of requesting permissions and not the library itself. Am I wrong ?

Reply:
Yea, the library does request for any permissions it uses beforehand.

So currently that would require explicitly informing the developer that the storage permission is required and that this should be included in the runtime permissions ask.

Reply:
I saw that permissions are requested in the MapActivity class in the library. When this activity is used ? The Permission class is used in the samples activities / or checks are done in the KujakuMapView class. But when permissions are requested in the library ? 

Could you please guide me ? 

Thanks

Reply:
https://github.com/onaio/kujaku/blob/master/library/src/main/java/io/ona/kujaku/views/KujakuMapView.java#L922

Reply:
I've added the check for the storage permission in the KujakuMapView